packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/ViewDiagramDescriptionConverter.java

```diff
@@ -308,17 +309,37 @@ private boolean matches(AQLInterpreter interpreter, String condition, VariableMa
 
     private List<ToolSection> createToolSections(ViewDiagramDescriptionConverterContext converterContext) {
         var capturedConvertedNodes = Map.copyOf(converterContext.getConvertedNodes());
+        var nodeCreationTools = this.getNodeTools(converterContext, capturedConvertedNodes);
+        var edgeTools = this.getEdgeTools(converterContext, capturedConvertedNodes);
+
+        // @formatter:off
+        var nodeCreationToolSection = ToolSection.newToolSection(UUID.randomUUID().toString())
+                .label(NODE_CREATION_TOOL_SECTION)
+                .tools(nodeCreationTools)
+                .imageURL("") //$NON-NLS-1$
+                .build();
+        var edgeCreationToolSection = ToolSection.newToolSection(UUID.randomUUID().toString())
+                .label(EDGE_CREATION_TOOL_SECTION)
+                .tools(edgeTools)
+                .imageURL("") //$NON-NLS-1$
+                .build();
+        return List.of(nodeCreationToolSection, edgeCreationToolSection);
+        // @formatter:on
+    }
+
+    private List<ITool> getNodeTools(ViewDiagramDescriptionConverterContext converterContext, Map<org.eclipse.sirius.components.view.NodeDescription, NodeDescription> capturedConvertedNodes) {
         List<ITool> nodeCreationTools = new ArrayList<>();
         for (var nodeDescription : converterContext.getConvertedNodes().keySet()) {
             List<NodeDescription> allTargetDescriptions = this.getAllTargetDescriptions(nodeDescription, converterContext);
+            String imageURL = this.viewToolImageProviderFactory.getToolImageProvider(nodeDescription).get();
```

Comment:
The tool image provider should be a stateless service and you should call `get(nodeDescription)` instead